functions/src/modules/users.functions.ts

```diff
@@ -0,0 +1,212 @@
+/////////////////////////////////////////////////
+//          Import Registered Users from a Spreadsheet(Http Triggered)
+//
+//      1. Parses a google spreadsheet
+//      2. Looks for two sheet --> Registrations
+//      3. Loads their data into the equivalent Firebase database paths
+/////////////////////////////////////////////////
+
+// Imports
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import { google } from 'googleapis';
+
+const db = admin.database();
+
+export const importUserRegistrationData = functions.https.onRequest(
+    async (req, res) => {
+        // Default 400 Bad Request Response
+        const badRequest = {
+            status: 400,
+            res: {
+                message:
+                    'Please define the sheet id, e.g sheet={SpreadSheet ID}',
+                status: false
+            }
+        };
+        // Grab the SpreadSheet ID from the Query Parameters
+        const spreadSheetId = req.query.sheet ? req.query.sheet : null;
+        if (!spreadSheetId) {
+            res.status(badRequest.status).send(badRequest.res);
+        } else {
+            performUserSpreadSheetImport(spreadSheetId).then(response => {

```

Comment:
I made this to prevent the function from getting too nested. I separated the main functionality, from the main function. But still, I think its a better idea to merge these functions because it is only called once in its lifetime. I appreciate your idea, I'll try to keep that in mind in the future.